### ISSUE TITLE
Empty body rule ignored

### ISSUE THREAD

**Issue by MClondi** [mockoon/mockoon#1628]
**Describe the bug**

When a HTTP request contains "Content-Type: application/json" header, the rule to detect empty request bodies is ignored. Mockoon shows a warning:
![image](https://github.com/user-attachments/assets/ded2eb8a-bc0c-4095-8633-cd7d2c0fb2ab)
This issue is not present on v9.0.0.
I think this might be related to the recent change discussed [here](https://github.com/mockoon/mockoon/issues/1595)

**Steps to Reproduce**

1. Create a POST route with a default response (no rules)
2. Add a 2nd response with a rule:
![image](https://github.com/user-attachments/assets/393c420f-b46a-41b0-9030-2c59787d9fd1)
3. Make a HTTP request with empty body and the following header: Content-Type: application/json -> Mockoon return response 1
4. Remove the Content-Type header -> Mockoon returns response 2

**Mockoon version:**

9.1.0

**OS / OS version:**

Ubuntu 24.04.1

**Comment by 255kb** [comment]
Indeed, thanks for catching this. I thought this would have been properly covered by the tests.

**Comment by BogdanMaier** [comment]
The same issue happens for a rule with regex targeting the entire body of the request:
"""
"rules": [
            {
              "target": "body",
              "modifier": "",
              "value": "filename=\".*some-file-name.*\"",
              "invert": false,
              "operator": "regex"
            }
          ],
"""

**Comment by 255kb** [comment]
📦 This was fixed in [v9.2.0](https://mockoon.com/releases/9.2.0/)
